### PR TITLE
Provide option to delete server-side study data during opt-out

### PR DESCRIPTION
**Pull request by cburkert**

### CODE REVIEW

**Comment by cburkert** [comment]
@fapdash Could you please have a look if everything sounds reasonable.
I've tested it with Chrome and Firefox.

File: app/pages/options.html [hunk 1]
```diff
@@ -29,15 +29,28 @@
...
           </p>
-          <span id="purgeCont" class="button btn-red" role="button">Delete data and opt-out</span>
-          <span id="purgeAbort" class="button btn-gray" role="button">Opt-out later</span>
+          <p id="deleteServer-box">
+            <label for="deleteServer">Remove data from study</label>
+            <input type="checkbox" id="deleteServer" name="deleteServer">
+          </p>
+          <p id="action-box">
+            <span id="purgeCont" class="button btn-red" role="button">Delete and opt-out</span>
+            <span id="purgeAbort" class="button btn-gray" role="button">Opt-out later</span>
```
Comment (fapdash) [inline app/pages/options.html]: Might sound a bit defensive? "Don't opt-out (now)" might do?

**Comment by fapdash** [comment]
@cburkert I don't get any error message if I try to delete the data and I have no internet connection.
The opt-out goes through so I can't delete the data anymore as I'm already opted out and the UUID has been deleted.

File: app/pages/options.html (hunk 1 above)
Comment (cburkert) [reply app/pages/options.html]: Fine by me. I changed it.

**Comment by cburkert** [comment]
@fapdash I cannot reproduce that. If I opt-out (with deletion) with no internet connection, I get the following error displayed. Which is expected because the `fetch` promise then fails and gets caught.

opt-out failure network

Are you sure you had the deletion checkbox checked and a report was submitted before? Otherwise a delete request is not attempted. Or did you maybe use the test server over loopback?

**Comment by fapdash** [comment]
@cburkert Oh, my bad. I didn't see that you where checking if data already had been sent..

**Comment by fapdash** [comment]
LGTM

**Pull request merged**